gemfirexd/tools/src/dunit/java/com/pivotal/gemfirexd/transactions/SnapshotTxGIIDUnit.java

```diff
@@ -336,6 +335,64 @@ public void run() {
     });
   }
 
+  public void testSnapshotGII_noGIIOption() throws Exception {
+
+    startVMs(0, 2);
+    Properties props = new Properties();
+    final Connection conn = TestUtil.getConnection(props);
+    conn.createStatement();
+
+    VM server1 = this.serverVMs.get(0);
+    VM server2 = this.serverVMs.get(1);
+    stopVMNums(-2);
+
+    server1.invoke(SnapshotTxGIIDUnit.class, "createPR", new Object[]{regionName, 2, 1});
+
+    // Put some values to initialize buckets
+    server1.invoke(new SerializableRunnable() {
+      @Override
+      public void run() {
+        putSomeValues(100);
+      }
+    });
+
+    blockGII(-2, InitialImageOperation.GIITestHookType.NoGIITrigger);
+
+    blockGII(-2, InitialImageOperation.GIITestHookType.AfterCalculatedUnfinishedOps);
+
+    restartServerVMNums(new int[]{2}, 0, null, null);
+
+    server2 = this.serverVMs.get(1);
+
+
+    // Create the PR region
+    server2.invoke(SnapshotTxGIIDUnit.class, "createPR", new Object[]{regionName, 2, 1});
+
+    waitForGIICallbackStarted(-2, InitialImageOperation.GIITestHookType.AfterCalculatedUnfinishedOps);
+
+    unblockGII(-2, InitialImageOperation.GIITestHookType.AfterCalculatedUnfinishedOps);
+
+    unblockGII(-2, InitialImageOperation.GIITestHookType.NoGIITrigger);
+
+
+    server1.invoke(new SerializableRunnable() {
+      @Override
+      public void run() {
+        putSomeValues(500);
+      }
+    });
+
+    server1.invoke(new SerializableRunnable() {
+      @Override
+      public void run() {
+        final GemFireCacheImpl cache = GemFireCacheImpl.getInstance();
+        final Region r = cache.getRegion(regionName);
+        assertEquals("After GII request failed for region , " +
+            "put not successful", 500, r.size());
```

Comment:
Nice to have will be to add one more server and redundancy and check that the gii actually succeeds after failing from one.